code/modules/cm_marines/dropship_ammo.dm

```diff
@@ -292,6 +292,97 @@
 	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), impact, 300, 40, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, create_cause_data(initial(name)), source_mob), 0.5 SECONDS) //Your standard HE splash damage rocket. Good damage, good range, good speed, it's an all rounder
 	QDEL_IN(src, 0.5 SECONDS)
 
+/obj/structure/ship_ammo/rocket/custom_missile
+	name = "\improper AIM-224B-C 'Widowmaker Custom'"
+	desc = "A modified version of the AIM-224B missile, allows for custom reagent mix to be inserted in the receptacle"
+	icon_state = "custom_missile"
+	ammo_name = "rocket"
+	travelling_time = 40
+	point_cost = 350
+	fire_mission_delay = 0 //direct bombard only
+	/// the limits of the custom missile
+	var/list/reaction_limits = list( "max_ex_power" = 300, "base_ex_falloff" = 60, "max_ex_shards" = 128,
+							"max_fire_rad" = 8, "max_fire_int" = 60, "max_fire_dur" = 48,
+							"min_fire_rad" = 3, "min_fire_int" = 5, "min_fire_dur" = 5
+	)
+	/// current assembly state of the missile
+	var/assembly_stage = ASSEMBLY_UNLOCKED
+	/// the maximum volume allowed in the rocket
+	var/max_container_volume = 300
+	/// the current total volume
+	var/current_container_volume = 0
+	/// the containers inside the rocket
+	var/list/obj/item/reagent_container/glass/containers = list()
+	var/list/allowed_containers = list(/obj/item/reagent_container/glass/beaker, /obj/item/reagent_container/glass/bucket, /obj/item/reagent_container/glass/bottle, /obj/item/reagent_container/glass/bucket/mopbucket)
+
+/obj/structure/ship_ammo/rocket/custom_missile/attackby(obj/item/item as obj, mob/user as mob)
+	if(HAS_TRAIT(item, TRAIT_TOOL_SCREWDRIVER))
+		if(assembly_stage == ASSEMBLY_UNLOCKED)
+			to_chat(user, SPAN_NOTICE("You lock the [name]."))
+			playsound(loc, 'sound/items/Screwdriver.ogg', 25, 0, 6)
+			assembly_stage = ASSEMBLY_LOCKED
+			icon_state = "custom_missile_ready"
+		else if(assembly_stage == ASSEMBLY_LOCKED)
+			to_chat(user, SPAN_NOTICE("You unlock the [name]."))
+			playsound(loc, 'sound/items/Screwdriver.ogg', 25, 0, 6)
+			assembly_stage = ASSEMBLY_UNLOCKED
+			icon_state = "custom_missile"
+	else if(is_type_in_list(item, allowed_containers) && (!assembly_stage || assembly_stage == ASSEMBLY_UNLOCKED))
+		if(current_container_volume >= max_container_volume)
+			to_chat(user, SPAN_DANGER("The [name] can not hold more containers."))
+			return
+		else
```

Comment:
you return on the line above, you dont need an else here.(this one for sure brothers)